src/backend/distributed/deparser/deparse_schema_stmts.c

```diff
@@ -53,6 +83,61 @@ DeparseAlterSchemaRenameStmt(Node *node)
 }
 
 
+static void
+AppendCreateSchemaStmt(StringInfo buf, CreateSchemaStmt *stmt)
+{
+	if (stmt->schemaElts != NIL)
+	{
+		elog(ERROR, "schema creating is not supported with other create commands");
+	}
+
+	if (stmt->schemaname == NULL)
+	{
+		elog(ERROR, "schema name should be specified");
+	}
+
+	appendStringInfoString(buf, "CREATE SCHEMA ");
+
+	if (stmt->if_not_exists)
+	{
+		appendStringInfoString(buf, "IF NOT EXISTS ");
+	}
+
+	appendStringInfo(buf, "%s ", quote_identifier(stmt->schemaname));
+
+	if (stmt->authrole != NULL)
+	{
+		appendStringInfo(buf, "AUTHORIZATION %s", RoleSpecString(stmt->authrole, true));
+	}
+}
+
+
+static void
+AppendDropSchemaStmt(StringInfo buf, DropStmt *stmt)
+{
+	Assert(stmt->removeType == OBJECT_SCHEMA);
+	Assert(list_length(stmt->objects) == 1);
+
+	appendStringInfoString(buf, "DROP SCHEMA ");
+
+	if (stmt->missing_ok)
+	{
+		appendStringInfoString(buf, "IF EXISTS ");
+	}
+
+	appendStringInfo(buf, "%s", quote_identifier(strVal(linitial(stmt->objects))));
```

Comment:
I know you only implemented this for deparsing 1 schema name before, see my comment there.

Even if there is no a good reason to disallow dropping more than 1 schema at a time I'd really love to see this future proved to add all schema's in the statement.

Statement deparsing is reused from many locations which assume that they either work correctly or have Asserts that we only expect 1 schema here.

As implementing correct deparsing is trivial please do so,

```suggestion
     foreach_ptr(...)
     {
              if (!first) appendStringInfoString(buf, ", ");
              appendStringInfoString(buf, quote_identifier(strVal(current_element)));
     }
```